src/common/DOMWorld.ts

```diff
@@ -284,17 +284,10 @@ export class DOMWorld {
     const watcher = new LifecycleWatcher(
       this.#frameManager,
       this.#frame,
-      waitUntil,
-      timeout
+      timeout,
+      waitUntil
     );
-    const error = await Promise.race([
-      watcher.timeoutOrTerminationPromise(),
-      watcher.lifecyclePromise(),
-    ]);
-    watcher.dispose();
```

Comment:
do we still need dispose?

Reply:
No, all watchers dispose themselves on error or return.